networkit/cpp/distance/CommuteTimeDistance.cpp

```diff
@@ -11,12 +11,14 @@
 #include <networkit/auxiliary/Log.hpp>
 #include <networkit/auxiliary/Timer.hpp>
 #include <networkit/distance/CommuteTimeDistance.hpp>
+#include <networkit/graph/GraphTools.hpp>
 
 namespace NetworKit {
 
 CommuteTimeDistance::CommuteTimeDistance(const Graph& G, double tol): Algorithm(), G(&G), tol(tol), lamg(1e-5) {
     // main purpose of method: preparing LAMG
-
+    if(G.isDirected()) WARN("Commute time distance is only supported for undirected graphs.");
```

Comment:
Instead of printing a warning, it might be better to `throw` a `std::runtime_error`.